contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py

```diff
@@ -0,0 +1,157 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import requests
+
+from opencensus.common.transports import sync
+from opencensus.ext.azure.protocol import Data
+from opencensus.ext.azure.protocol import Envelope
+from opencensus.ext.azure.protocol import RemoteDependency
+from opencensus.ext.azure.protocol import Request
+from opencensus.ext.azure import utils
+from opencensus.trace import base_exporter
+from opencensus.trace import execution_context
+from opencensus.trace.span import SpanKind
+
+
+class AzureExporter(base_exporter.Exporter):
+    """An exporter that sends traces to Microsoft Azure Monitor.
+
+    :type config: dict
+    :param config: Configuration for the exporter. Defaults to None.
+
+    :type transport: :class:`type`
+    :param transport: Class for creating new transport objects. It should
+                      extend from the base_exporter :class:`.Transport` type
+                      and implement :meth:`.Transport.export`. Defaults to
+                      :class:`.SyncTransport`. The other option is
+                      :class:`.AsyncTransport`.
+    """
+
+    def __init__(self, config=None, transport=sync.SyncTransport):
+        self.config = config or utils.Config()
+        self.transport = transport(self)
+
+    def span_data_to_envelope(self, sd):
+        # print('[AzMon]', sd)
+        # print('trace_id:', sd.context.trace_id)
+        # print('tracestate:', sd.context.tracestate)
+        # print('span_id:', sd.span_id)
+        # print('parent_span_id:', sd.parent_span_id)
+        # print('attributes:', sd.attributes)
+        # print('start_time:', sd.start_time)
+        # print('end_time:', sd.end_time)
+        envelope = Envelope(
+            iKey=self.config.instrumentation_key,
+            tags=dict(utils.azure_monitor_context),
+            time=sd.start_time,
+        )
+        envelope.tags['ai.operation.id'] = sd.context.trace_id
+        if sd.parent_span_id:
+            envelope.tags['ai.operation.parentId'] = '|{}.{}.'.format(
+                sd.context.trace_id,
+                sd.parent_span_id,
+            )
+        if sd.span_kind == SpanKind.SERVER:
+            envelope.name = 'Microsoft.ApplicationInsights.Request'
+            data = Request(
+                id='|{}.{}.'.format(sd.context.trace_id, sd.span_id),
+                duration=utils.timestamp_to_duration(
+                    sd.start_time,
+                    sd.end_time,
+                ),
+                responseCode='0',  # TODO
+                success=True,  # TODO
```

Comment:
As far as I know of, the operation is said to be successful (in Azure Monitor) if the span completed without generating a runtime exception. @reyang what are your thoughts about it, in order to complete this TODO.

Reply:
I don't know at this moment, that's why I put TODO here.